### PR TITLE
test: fix failing MAAS UI e2e tests

### PR DESCRIPTION
**Pull request by Jay-Topher**
## Done
- Add script to ensure successful image sync before test run
- Update failing footer tests for legacy footer removal

## QA steps

- [ ] Ensure cypress CI runs succcessfully

## Fixes
[1967](https://warthogs.atlassian.net/jira/software/c/projects/MAASENG/issues/MAASENG-1967)
Fixes:

### CODE REVIEW

File: cypress/support/commands.ts [hunk 1]
```diff
@@ -44,7 +44,9 @@ Cypress.Commands.add("addMachine", (hostname = generateName()) => {
   cy.get("input[name='hostname']").type(hostname);
   cy.get("input[name='pxe_mac']").type(generateMac());
   cy.get("select[name='power_type']").select("manual").blur();
-  cy.get("button[type='submit']").click();
+  cy.get("button[type='submit']", { timeout: LONG_TIMEOUT })
+    .should("be.enabled")
+    .click();
```
Comment (petermakowski) [inline cypress/support/commands.ts:49]: As explained in the acceptance criteria, our goal here is to:

> Wait for ephemeral OS to be available or provide a workaround allowing the test to pass when it’s unavailable

Generic timeout wait specific enough. I'd recommend to talk to someone from the MAAS core team, share a screenshot of the failing test where ephemeral message is visible and find if there are better ways of detecting this.

Comment (Jay-Topher) [reply cypress/support/commands.ts:49]: Okay, thanks I reach out to someone concerning this, this is not up for review yet, I created a draft PR as a sandbox to get Cypress test results.

File: .github/workflows/cypress.yml [hunk 2]
```diff
@@ -36,6 +36,7 @@ jobs:
         run: |
           export API_KEY=`sudo maas apikey --username=admin`
           maas login admin http://localhost:5240/MAAS $API_KEY
+          while [ $(maas admin boot-resources is-importing | cat) == "true" ]; do sleep 10; done; echo "syncing finished"
```
Comment (petermakowski) [inline .github/workflows/cypress.yml:39]: I think ideally it should be moved to a separate step (e.g. with a name "Wait for MAAS boot resources") which would help track its progress (and execution time).

**Review by petermakowski — CHANGES_REQUESTED** [review]
Glad you managed to figure this out! Let's clean this up a bit before we merge.

**Review by petermakowski — APPROVED** [review]

**Pull request merged**